### PR TITLE
[gosrc2cpg] - FieldAccess call node handling along with Receiver identifier handling for method call node.

### PR DESCRIPTION
**Pull request by pandurangpatil**
1. Receiver argument handling for method call nodes
2. Fixed the duplicate call nodes getting created for Constructor call nodes.
3. Few additional unit tests
4. FieldAccess call node handling along with field identifier.
5. Created `NotHandledType` for the ast types which we have not yet handled and logged the warning.
6. Some other code refactoring.

### CODE REVIEW

File: joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/FieldAccessTests.scala [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        |func foo() {
+        |   var a Person = Person{fname: "Pandurang", lname: "Patil"}
+        |   println(a.fname)
+        |}
+        |""".stripMargin)
+
+    "External Field access test" in {
+      val List(fieldAccessCall) = cpg.method("main").astChildren.fieldAccess.l
+      fieldAccessCall.name shouldBe Operators.fieldAccess
+      // TODO: We need to find and set the right full name
```
Comment (DavidBakerEffendi) [inline joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/FieldAccessTests.scala:35]: Field access methods are typically the best way to resolve the type once the AST is created: https://github.com/joernio/joern/blob/master/semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/operatorextension/nodemethods/FieldAccessMethods.scala

File: joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/TypeDeclMethodCallTests.scala [hunk 2]
```diff
@@ -45,5 +45,14 @@ class TypeDeclMethodCallTests extends GoCodeToCpgSuite {
       x.typeFullName shouldBe "string"
       x.dispatchType shouldBe DispatchTypes.STATIC_DISPATCH
     }
+
+    "check this/receiver argument is correctly getting passed" in {
+      cpg.call("fullName").argument.size shouldBe 1
+      val List(x) = cpg.call("fullName").argument.l
```
Comment (DavidBakerEffendi) [inline joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/TypeDeclMethodCallTests.scala:51]: You can save yourself the effort of checking types later down the line with
```suggestion
      val List(x: Identifier) = cpg.call("fullName").argument.l: @unchecked
```

File: joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/MethodCallTests.scala [hunk 3]
```diff
@@ -456,6 +456,15 @@ class MethodCallTests extends GoCodeToCpgSuite(withOssDataflow = true) {
       x.name shouldBe "bar"
       x.isExternal shouldBe true
     }
+
+    "Check arguments nodes" in {
+      cpg.call("bar").argument.size shouldBe 2
+      val List(a, b) = cpg.call("bar").argument.l
```
Comment (DavidBakerEffendi) [inline joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/MethodCallTests.scala:462]: Similarly here ```suggestion
      val List(a: Identifier, b: Identifier) = cpg.call("bar").argument.l: @unchecked
```

File: joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/TypeDeclMethodCallTests.scala (hunk 2 above)
Comment (pandurangpatil) [reply joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/TypeDeclMethodCallTests.scala]: done

File: joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/MethodCallTests.scala (hunk 3 above)
Comment (pandurangpatil) [reply joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/MethodCallTests.scala]: done

File: joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/FieldAccessTests.scala (hunk 1 above)
Comment (DavidBakerEffendi) [reply joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/FieldAccessTests.scala:35]: @pandurangpatil will explain in the call

**Review by DavidBakerEffendi — APPROVED** [review]

**Pull request merged**